emu_mps/observables.py

```diff
@@ -0,0 +1,40 @@
+from pulser.backend.state import State
+from pulser.backend.observable import Observable
+from emu_mps.custom_callbacks import calculate_entanglement_entropy
+from emu_mps.mps import MPS
+from typing import Sequence, Any
+
+
+class EntanglementEntropy(Observable):
+    """Entanglement Entropy subclass used only in emu_mps"""
+
+    def __init__(
+        self,
+        bond_index: int,
+        *,
+        evaluation_times: Sequence[float] | None = None,
+        tag_suffix: str | None = None,
+    ):
+        super().__init__(evaluation_times=evaluation_times, tag_suffix=tag_suffix)
+        self.bond_index = bond_index
+
+    @property
+    def _base_tag(self) -> str:
+        return "entanglement_entropy"
+
+    def _to_abstract_repr(self) -> dict[str, Any]:
+        repr = super()._to_abstract_repr()
+        repr["bond_index"] = self.bond_index
+        return repr
+
+    def apply(self, *, state: State, **kwargs: Any) -> float:
+        if not isinstance(state, MPS):
+            raise NotImplementedError(
+                "Entanglement entropy observable is only available for emu_mps emulator."
+            )
+        if not (0 < self.bond_index < len(state.factors)):
```

Comment:
Can you compute the entropy for bond i from MPS tensors of both sides?
If so what about indexing bonds in [[0, qubit_count - 2]]?
It feels more natural from computer perspective (Python lists indexing is 0-based), but maybe debatable from physics perspective...

Reply:
see also my comment above about the range of `b` values actually permitted in the code